### ISSUE TITLE
Inverse of logamplitude?

### ISSUE THREAD

**Issue by carlthome** [librosa/librosa#421]
librosa.logamplitude is great! How about also having an inverse available? E.g. something like:

``` Python
def linearamplitude(log_S, ref_power=1.0)
    """Linear-scale the amplitude of a spectrogram."""
    return 10 ** (log_S / 10 + np.log10(ref_power))
```

I guess it wouldn't be exact though, but still useful in some cases maybe?

**Comment by dpwe** [comment]
Haha.  We made a function to calculate dB of power and called it
logamplitude!

My preference would have been to call it amplitude_to_dB or something, and
define it as 20*log10(amplitude).

Then we could also provide dB_to_amplitude.

The function as defined should be called logpower, since the input is
assumed power, not amplitude.

If we're mapping between two representations, isn't it better to name the
input and output, rather than just the output?

  DAn.

**Comment by bmcfee** [comment]
> Haha.  We made a function to calculate dB of power and called it
> logamplitude!

"we" :grin: 

Yeah, I think what happened here is that the initial cut simply did 10log10 scaling without purporting to return dB or accept input as power. At some later point, we added the noise floor clipping parameter (`top_db`) which was implicitly in dB units even when the output was not, hence the potentially confusing name.  Isn't software fun?

I'd be okay with an API change / rename to have more explicit names for this functionality, as well as including the inverse transformation.

Meanwhile, what joker wrote this [crappy exception message](https://github.com/librosa/librosa/blob/master/librosa/core/spectrum.py#L691)?

**Comment by dpwe** [comment]
>
>
> Meanwhile, what joker wrote this crappy exception message
> <https://github.com/librosa/librosa/blob/master/librosa/core/spectrum.py#L691>
> ?
>

That's pretty weird.  There's nothing wrong with subtracting a negative
maximum.  So I don't think that exception should exist in the first place.

(is it just for me that the whitespace isn't displaying in a python-legal
way?)

  DAn.

**Comment by dpwe** [comment]
Oh, I misread the meaning of top_db.

I think the argument should be called db_range.  Then you can insist it's
greater than 0.

  DAn.

**Comment by carlthome** [comment]
I'M CONFUSED. :rabbit:

So this is why `logamplitude()` is called with S**2 throughout the docs? To make magnitudes into power? Always been a bit confused by this actually.

**Comment by dpwe** [comment]
On Wed, Oct 19, 2016 at 10:26 AM, Carl Thomé <notifications@github.com>
wrote:

> I'M CONFUSED. [image: 🐰]
>
> So this is why logamplitude() is called with S**2 throughout the docs? To
> make magnitudes into power? Always been a bit confused by this actually.
>
Exactly.

  DAn.

**Comment by bmcfee** [comment]
Before I go hacking on this, does this sound like a reasonable set of changes?

- `logamplitude` renamed to `power_to_db`; an alias included for backwards compatibility
- new function `amplitude_to_db`which just wraps `power_to_db(X**2)` 
- inverse functions `db_to_amplitude` and `db_to_power` defined accordingly

**Comment by dpwe** [comment]
SGTM.

  DAn.

**Comment by bmcfee** [comment]
Implemented as #464
